### ISSUE TITLE
scrolling awkwardly stops and jumps when new tiles are loaded

### ISSUE THREAD

**Issue by F3der1co** [ConstructFund/madewithconstructwebsite#15]
The scrolling of the site is a bit awkward as every time a new batch of games load the page stops and then jumps.
Maybe the main problem is just the scroll position jumping after the load instead of staying where it was, as it is disorienting to me currently.
it also seems like the tile position changes (i.e. tiles move by one or so, making them jump to different rows)

**Comment by fodi** [comment]
I love the content and the layout but the scrolling behaviour drives me up the wall as well. Why remove already loaded cards from the top anyway when scrolling down?

Also PageUp/PageDown/Home/End buttons don't seem to work in Firefox.

**Comment by skymen** [comment]
> Why remove already loaded cards from the top anyway when scrolling down?

This is done so the page doesn't have to load more and more games in memory. This clearly not an issue right now since there is only about 150 games, but as the list gets longer and especially as each item gets more info about it (if I revamp the website) I want it to keep running smoothly.

This is probably badly implemented on my end, but I don't have time to review my own code at the moment. I have a site revamp planned at some point in the future. I will rework the scrolling then (unless someone wants to do a PR which I will gladly accept)

**Comment by fodi** [comment]
I absolutely get the reasoning (keeping things smooth) but ironically the opposite is happening now. Unfortunately I don't know anything about Vue so I can't help with a PR, but are you using some sort of "virtual list" functionality? Because (either 1st or 3rd party) implementations usually exists for popular JS UI frameworks and I'd guess it would take care of the issue of scrolling very long lists without the current jankiness. Maybe one of these?

https://vueuse.org/core/useVirtualList/
https://github.com/Akryum/vue-virtual-scroller
https://github.com/tangbc/vue-virtual-scroll-list

**Comment by skymen** [comment]
yeah you're right I should do this using a virtual scroller, I don't remember why I didn't do it at the time but I remember looking into it and deciding against it. It was 3 years ago tho, so a rework today would use Vue 3 and have a properly implemented virtual scrolling system
